File: visual_basic/bucles/mostrar_menu_interactivo/mostrar_menu_interactivo.vb
Imports System
Module mostrar_menu_interactivo
    Sub Main(args As String())
        Dim contactos As New List(Of String)
        Dim opcion As Integer = 0
        Do
            Console.Clear()
            Console.WriteLine("=== AGENDA DE CONTACTOS ===")
            Console.WriteLine("1. Ver contactos")
            Console.WriteLine("2. Agregar contacto")
            Console.WriteLine("3. Eliminar contacto")
            Console.WriteLine("4. Salir")
            Console.Write("Seleccione una opción: ")
            opcion = Convert.ToInt32(Console.ReadLine())
            Select Case opcion
                Case 1
                    Console.WriteLine(vbCrLf & "Lista de contactos:")
                    If contactos.Count = 0 Then
                        Console.WriteLine("No hay contactos.")
                    Else
                        For Each c In contactos
                            Console.WriteLine("- " & c)
                        Next
                    End If
                    Console.ReadKey()
                Case 2
                    Console.Write(vbCrLf & "Ingrese nombre del contacto: ")
                    Dim nuevo As String = Console.ReadLine()
                    contactos.Add(nuevo)
                    Console.WriteLine("Contacto agregado.")
                    Console.ReadKey()
                Case 3
                    Console.Write(vbCrLf & "Ingrese contacto a eliminar: ")
                    Dim eliminar As String = Console.ReadLine()

                    If contactos.Contains(eliminar) Then
                        contactos.Remove(eliminar)
                        Console.WriteLine("Contacto eliminado.")
                    Else
                        Console.WriteLine("El contacto no existe.")
                    End If
                    Console.ReadKey()
                Case 4
                    Console.WriteLine("Saliendo del programa...")
                Case Else
                    Console.WriteLine("Opción inválida.")
                    Console.ReadKey()
            End Select
        Loop While opcion <> 4
    End Sub
End Module

File: python/mostrar_menu_interactivo.py
while True:
    print("\n--- Menú de Agenda ---")
    print("1. Registrar contacto")
    print("2. Mostrar contactos")
    print("3. Buscar contacto")
    print("4. Eliminar contacto")
    print("5. Salir")
    opcion = input("Seleccione una opción: ")
    if opcion == "1":
        print("Opción: Registrar contacto")
    elif opcion == "2":
        print("Opción: Mostrar contactos")
    elif opcion == "3":
        print("Opción: Buscar contacto")
    elif opcion == "4":
        print("Opción: Eliminar contacto")
    elif opcion == "5":
        print("Saliendo del menú...")
        break
    else:
        print("Opción no válida, intente de nuevo")